activerecord/test/cases/tasks/database_tasks_test.rb

```diff
@@ -119,12 +129,89 @@ def test_raises_an_error_if_no_migrations_have_been_made
       assert_not_predicate internal_metadata, :table_exists?
 
       assert_raises(ActiveRecord::NoEnvironmentInSchemaError) do
-        ActiveRecord::Tasks::DatabaseTasks.check_protected_environments!
+        ActiveRecord::Tasks::DatabaseTasks.check_protected_environments!("arunit")
       end
     ensure
       schema_migration.delete_version("1")
       internal_metadata.create_table
     end
+
+    private
+      def recreate_metadata_tables
+        schema_migration = ActiveRecord::Base.connection.schema_migration
+        schema_migration.drop_table
+        schema_migration.create_table
+
+        internal_metadata = ActiveRecord::Base.connection.internal_metadata
+        internal_metadata.drop_table
+        internal_metadata.create_table
+      end
+  end
+
+  if current_adapter?(:SQLite3Adapter) && !in_memory_db?
```

Comment:
I can look into this. The reason was because we override the database configuration in the `with_multi_db_configurations` helper, which specifies `adapter: sqlite3` 🤔

Reply:
Most other multi-database tests in this file actually stub out the actual logic so I have to put a bunch of effort in figuring out how to make a multi-database test configuration that works across all possible adapters.

Do we prefer if the test ran against all adapters but stubbed out the `check_current_protected_environment!` method? Or do we prefer running against only `sqlite3` but it ran through the whole flow and made stronger assertions?